src/internal/constants.ts

```diff
@@ -18,6 +18,9 @@ export const validOrigins = [
   'https://*.sharepointonline.com',
   'https://outlook.office.com',
   'https://outlook-sdf.office.com',
+  'https://retailservices-dev.teams.microsoft.com', // Teams: Tasks(Dev) app
+  'https://retailservices.teams.microsoft.com', // Teams: Tasks app
+  'https://tasks.office.com', // Planner
```

Comment:
I'm adding an API to send a message from parent to child. In my scenario parent is Tasks app in Teams, and child is Planner webapp iframed inside the Tasks app.
The processMessage() function only processes messages if they are from a valid origin. IF a message is from a valid origin, updateRelationships() will go ahead and assign GlobalVars.childWindow and GlobalVars.childOrigin. 

In my scenario, Tasks app will receive an "initialize" message from it's child: Planner iframe (tasks.office.com) and only allow that message process and save it as child if its domain is in valid domains list. Therefore, we need both parent and child, the sources of messages to be in valid domains so that Teams SDK stores them as parent and child respectively in the respective windows.

Reply:
Still, we shouldn't be putting any app-specific URLs in here. I think we need to add a new method or extend the existing `initialize` method to take a list of `additionalValidOrigins` that the app expects to communicate with and have it be provided dynamically at runtime.

Reply:
Agreed. That's a better sdk/platform feature.